tests/test_http_monitor.py

```diff
@@ -0,0 +1,105 @@
+"""Test suite for monitoring_scripts.http_monitor"""
+
+import pytest
+import requests
+
+from monitoring_scripts import http_monitor
+from monitoring_scripts.nagios_common import Codes
+
+
+def assert_nagios_exit(expected_code, exit_code, stdout):
+    """Common assertions to test Nagios compatibility"""
+
+    assert exit_code == expected_code.value
+    assert stdout.startswith(expected_code.name + ':')
+
+
+def run_and_assert(capfd, expected_code=Codes.OK, url='http://foo.bar', timeout=30,
+                   redirect_unknown=True, debug=False):
+    """Run http_monitor.main with the specified parameters and asserts that its output and return
+    code match the expected return code and the Nagios guidelines """
+
+    with pytest.raises(SystemExit) as exit_info:
+        http_monitor.main(url, timeout, redirect_unknown, debug)
+
+    out = capfd.readouterr()[0]
+    assert_nagios_exit(expected_code, exit_info.value.code, out)
+
+
+def generate_response(mocker, status_code=200):
+    """Generates a fake requests.Response with the specified HTTP status code"""
+
+    response = mocker.Mock()
+    response.status_code = status_code
+
+    return response
+
+
+def invalid_url(url, capfd):
+    """Assert UNKNOWN status with wrong URL"""
+
+    run_and_assert(capfd, expected_code=Codes.UNKNOWN, url=url)
+
+
+def test_invalid_url_invalid_protocol(capfd):
+    """Assert UNKNOWN status with wrong URL protocol"""
+
+    url = 'htt://foo.bar'
+
+    invalid_url(url, capfd)
+
+
+def test_invalid_url_invalid_domain(capfd):
+    """Assert UNKNOWN status with wrong URL domain"""
+
+    url = 'http://foo'
+
+    invalid_url(url, capfd)
+
+
+def test_connection_timeout(mocker, capfd):
+    """Assert CRITICAL status if connection times out"""
+
+    mocker.patch('requests.head', side_effect=requests.ConnectTimeout())
+
+    run_and_assert(capfd, expected_code=Codes.CRITICAL)
+
+
+def test_response_timeout(mocker, capfd):
+    """Assert CRITICAL status if response times out"""
+
+    mocker.patch('requests.head', side_effect=requests.ReadTimeout())
+
+    run_and_assert(capfd, expected_code=Codes.CRITICAL)
+
+
+def test_200_ok(mocker, capfd):
```

Comment:
I'd rather have `test_nominal`, `test_unknown` and `test_critical` functions each parametrized with the return codes and related `http_monitor.main`'s kwargs.

Reply:
What would the `test_nominal` be?

Reply:
The content of test_200_ok, really, as there's only one return code where it's OK

Reply:
Future behavior might include an `OK` status code on 2xx HTTP response code

Reply:
Then that future behavior would be implemented in the `test_nominal` test, since AFAIC `OK` is nominal